spec/compiler/codegen/proc_spec.cr

```diff
@@ -761,4 +761,49 @@ describe "Code gen: proc" do
       f.call(20)
       )).to_i.should eq(21)
   end
+
+  it "can assign proc that returns anything to proc that returns nil (#3655)" do
+    run(%(
+      class Foo
+        @block : -> Nil
```

Comment:
Hmm. What are the corresponding rules around `@block : -> Void`?

Reply:
Same. `Void` in Proc return type is the same as `Nil`.